### PR TITLE
[FIX] web: fix autofocus when input type is 'number'

### PR DESCRIPTION
**Pull request by xavierbol**
Before this commit, when the autofocus is used on an input with
`type="number"`, the `selectionEnd` is not available because the input
element's type ('number') does not support selection.

This commit checks the input type is not a number:
- if it is the case then the selection is used to place the cursor at
  the end of the input content,
- otherwise the selection properties are not used and the input is
  simply focused.

### CODE REVIEW

**Review by sdegueldre — APPROVED** [review]

**Comment by sdegueldre** [comment]
@robodoo r+

**Comment by sdegueldre** [comment]
@xavierbol you forgot to import xml from owl

**Comment by xavierbol** [comment]
Oh indeed, sorry, I didn't see it was different than master. 😬 
I will also destroy the component, like you did in others tests. 🙂

File: addons/web/static/tests/core/utils/hooks_tests.js [hunk 1]
```diff
@@ -42,6 +42,34 @@ QUnit.module("utils", () => {
...
+                    <input type="number" t-ref="autofocus" />
+                </span>
+            `;
+
+            registry.category("services").add("ui", uiService);
+
+            const env = await makeTestEnv();
+            const target = getFixture();
+            const comp = await mount(MyComponent, { env, target });
+            await nextTick();
```
Comment (aab-odoo) [inline addons/web/static/tests/core/utils/hooks_tests.js:62]: not necessary I think

Comment (xavierbol) [reply addons/web/static/tests/core/utils/hooks_tests.js]: Indeed, it is not necessary. 🙂

**Review by sdegueldre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**